### ISSUE TITLE
Duplicate games can appear when loading several pages of entries

### ISSUE THREAD

**Issue by mkalam-alami** [mkalam-alami/ludumdare-feedback-friends#6]
Due to caching, the 1st page of entries can have a different age from the following pages ; because of that, some entries can appear twice in different pages.

At first I didn't think it would be too much of an issue, but it actually seems to occur quite frequently if people have been recently commenting or submitting games on LD. Example (see entries 9 and 10):

![clipboard02](https://cloud.githubusercontent.com/assets/694508/17245821/abcb9c44-5588-11e6-943c-b3226b924261.jpg)

EDIT: Possible fixes:

* Synchronize TTLs for different pages of a same search *(but it can still show dupes)*
* Remove caching
* ?

**Comment by ttencate** [comment]
Ah, one of the two hard problems in computer science... Do you have metrics on how much the cache is helping, or did you just put it in as a precaution?

Could it work to remove the caching from the PHP code entirely, and just stick a Varnish server in front? That would simplify the code a lot. Varnish is super speedy, so it might take enough load off the server that we can use a short TTL like 10 seconds.

**Comment by mkalam-alami** [comment]
Yeah maybe a 2mn TTL is just too much.

I've never used Varnish but why not give it a try, it will certainly improve performance even more.

FYI I did run some load testing: cached pages are served about 4 times faster (from ~140 to 35ms). IIRC, thanks to that the VM handles about 80 incoming requests per second (keeping >200 Apache connections busy). Without the APCu caching, MySQL maxes out the CPU way before that. I also have Cloudflare set up for static resources.

 I've just set the APCu TTL to 10s instead, I'll see if it impacts performance or not.

See you on IRC maybe to discuss the Varnish option!
